internal/product/service.go

```diff
@@ -7,15 +7,15 @@ import (
 )
 
 type productDBAccessor interface {
-	GetProductsByVendor(ctx context.Context, vendorID string) ([]Product, error)
+	GetProductsByVendor(ctx context.Context, spec GetProductsByVendorSpec) ([]Product, error)
 }
 
 type ProductService struct {
 	productDBAccessor
 }
 
-func (p *ProductService) GetProductsByVendor(ctx context.Context, vendorID string) ([]Product, error) {
-	return p.productDBAccessor.GetProductsByVendor(ctx, vendorID)
+func (p *ProductService) GetProductsByVendor(ctx context.Context, spec GetProductsByVendorSpec) ([]Product, error) {
```

Comment:
```suggestion
func (p *ProductService) GetProductsByVendor(ctx context.Context, vendorID, spec GetProductsByVendorSpec) ([]Product, error) {
```

I think we should just keep the `vendorID` field. It enforces the need for `vendorID` for every client verbosely